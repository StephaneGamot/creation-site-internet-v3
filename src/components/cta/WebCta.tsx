"use client"

import ContactButton from "../Buttons/ContactButton"
import { useTranslations } from "next-intl"

export default function WebCta() {
  const t = useTranslations("webCta");

  return (
    <section className="text-center">
      <h2 className="text-3xl font-title font-semibold mb-4">
        {t("title")}
      </h2>
      <p className="text-lg text-gray-700 mb-6">
        {t("text")}
      </p>
      <ContactButton />
    </section>
  );
}
