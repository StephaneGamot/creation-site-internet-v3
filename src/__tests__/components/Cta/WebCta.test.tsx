import React from "react";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import WebCta from "@/components/cta/WebCta";

// ✅ Mock de next-intl
jest.mock("next-intl", () => ({
  useTranslations: () => {
    return (key: string) => {
      if (key === "title") return "Un site web à votre image";
      if (key === "text") return "Conception sur mesure, performance, SEO... tout y est.";
      if (key === "aria") return "Aller à la page de contact";
      if (key === "label") return "Me contacter";
      return "";
    };
  },
  useLocale: () => "fr",
}));

// ✅ Mock du bouton ContactButton
jest.mock("@/components/Buttons/ContactButton", () => () => (
  <a href="/fr/contact" role="button">Me contacter</a>
));

describe("WebCta", () => {
  it("affiche le titre et le texte", () => {
    render(<WebCta />);
    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent("Un site web à votre image");
    expect(screen.getByText("Conception sur mesure, performance, SEO... tout y est.")).toBeInTheDocument();
  });

  it("affiche le bouton de contact", () => {
    render(<WebCta />);
    const button = screen.getByRole("button", { name: "Me contacter" });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute("href", "/fr/contact");
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<WebCta />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const { asFragment } = render(<WebCta />);
    expect(asFragment()).toMatchSnapshot();
  });
});
