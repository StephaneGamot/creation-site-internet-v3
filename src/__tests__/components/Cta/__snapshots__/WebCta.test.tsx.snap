// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`WebCta correspond au snapshot 1`] = `
<DocumentFragment>
  <section
    class="text-center"
  >
    <h2
      class="text-3xl font-title font-semibold mb-4"
    >
      Un site web Ã  votre image
    </h2>
    <p
      class="text-lg text-gray-700 mb-6"
    >
      Conception sur mesure, performance, SEO... tout y est.
    </p>
    <a
      href="/fr/contact"
      role="button"
    >
      Me contacter
    </a>
  </section>
</DocumentFragment>
`;
